### PR TITLE
Save toggler status

### PR DESCRIPTION
**Pull request by hostedposted**
## Description
So if you click Toggle Membership and Disable Monster Encounters then reload prodigy, Toggle Membership and Disable Monster Encounters will be auto clicked.

### CODE REVIEW

**Comment by PatheticMustan** [comment]
clearly this isn't ready, so just request a review when it's done

**Comment by ChickenOnYT** [comment]
hostedposted isnt a collab

**Comment by Alexander1907297** [comment]
stfu u stupid bitch

**Comment by BoredFishRE** [comment]
what the fuck don't be rude

**Comment by ChickenOnYT** [comment]
this fucking sentence is the third one with a cuss word

**Comment by Alexander1907297** [comment]
i was playing i was playing my bad i was playing god i was finna see what youll was finna say

**Comment by BoredFishRE** [comment]
that wasn't a sentence

**Comment by ChickenOnYT** [comment]
> that wasn't a sentence

my 3rd grader teacher says a sentence needs at least 3 words to be a sentence.

**Comment by BoredFishRE** [comment]
> 
> 
> > that wasn't a sentence
> 
> my 3rd grader teacher says a sentence needs at least 3 words to be a sentence.

in that logic "dfjkhfsdhkj fsdkhjsdfyuigse dfshjkgdfskjhgdfsj" is a sentence. A sentence has to make sense lmfao.

**Comment by CyberXVI** [comment]
Yeah this totally went off topic so fast
It's best if you just closed the issue

**Comment by BoredFishRE** [comment]
no

**Comment by ChickenOnYT** [comment]
> > > that wasn't a sentence
> > 
> > 
> > my 3rd grader teacher says a sentence needs at least 3 words to be a sentence.
> 
> in that logic "dfjkhfsdhkj fsdkhjsdfyuigse dfshjkgdfskjhgdfsj" is a sentence. A sentence has to make sense lmfao.

well it has to be words, those arent words, just letters. yeah it has to make sense but "stfu u stupid bitch" makes sense even if its rude, everything here makes sense but is just cussing and rudeness lmao

**Comment by ChickenOnYT** [comment]
> > > that wasn't a sentence
> > 
> > 
> > my 3rd grader teacher says a sentence needs at least 3 words to be a sentence.
> 
> "I am" is a sentence

smart

**Comment by ChickenOnYT** [comment]
> Yeah this totally went off topic so fast
> It's best if you just closed the issue

its not even an issue

**Comment by ChickenOnYT** [comment]
> i was playing i was playing my bad i was playing god i was finna see what youll was finna say

but the I'm is a yeah

**Comment by ChickenOnYT** [comment]
i made it more famiwly fwiendly™
image

File: willsCheatMenu/src/index.ts [hunk 1]
```diff
@@ -106,8 +106,14 @@ export class Toggler extends Hack {
 		this.element.setAttribute("status", "false");
 		this.setClick(async () => {
 			this.status = !this.status;
-			if (this.status) await this.enabled?.();
-			else await this.disabled?.();
+			if (this.status) {
+				localStorage.setItem(name, "true")
```
Comment (PatheticMustan) [inline willsCheatMenu/src/index.ts:110]: This.name

File: willsCheatMenu/src/index.ts [hunk 2]
```diff
@@ -118,6 +124,9 @@ export class Toggler extends Hack {
 	}
 	setEnabled(event: () => unknown) {
 		this.enabled = event;
+		if (localStorage.getItem(this.name) === "true") {
```
Comment (PatheticMustan) [inline willsCheatMenu/src/index.ts:127]: This should be called in the constructor, not in setEnabled

Comment (hostedposted) [reply willsCheatMenu/src/index.ts:127]: If it is called in the constructor ``this.enabled`` will not be defined and the function will not execute.

Comment (PatheticMustan) [reply willsCheatMenu/src/index.ts:127]: no

and where it is now doesn't do anything

Comment (hostedposted) [reply willsCheatMenu/src/index.ts]: Okay fixed it.

**Comment by PatheticMustan** [comment]
Oh jesus, I'm a fool. Sorry for doubting you @hostedposted, you were right about the constructor setenabled thing.

**Pull request merged**